### PR TITLE
Maya-126871 Delete compound attributes does not remove their connections.

### PR DESCRIPTION
**Pull request by alicedegirolamo**
# Description

With this PR, when deleting a compound attribute, the associated connections are also deleted. 

Fixes [issue 126871](https://jira.autodesk.com/browse/MAYA-126871).

# How Has This Been Tested?

I updated the existing `testAttributes.py` adding the test `testRemoveCompoundAttribute`.

### CODE REVIEW

**Review by JGamache-autodesk — DISMISSED** [review]
You will most probably have to run clang-format on this one. I can see lines 491-493 are not to spec.

**Comment by alicedegirolamo** [comment]
> You will most probably have to run clang-format on this one. I can see lines 491-493 are not to spec.

I re-run the clang-format but it seems correct to the clang-format.

File: lib/mayaUsd/ufe/UsdAttributes.cpp [hunk 1]
```diff
@@ -455,6 +455,24 @@ bool UsdAttributes::canRemoveAttribute(const UsdSceneItem::Ptr& item, const std:
     }
     return false;
 }
+static void removeConnections(const PXR_NS::UsdPrim& prim, const SdfPath& srcPropertyPath)
```
Comment (seando-adsk) [inline lib/mayaUsd/ufe/UsdAttributes.cpp:458]: PXR_NS::SdfPath

File: test/lib/ufe/testAttributes.py [hunk 2]
```diff
@@ -183,6 +183,64 @@ def testAddRemoveAttribute(self):
         with self.assertRaisesRegex(KeyError, "Attribute 'MyAttribute' does not exist") as cm:
             attr = ball35Attrs.attribute("MyAttribute")
 
+    @unittest.skipIf(os.getenv('UFE_PREVIEW_VERSION_NUM', '0000') < '4024', 'Test for UFE preview version 0.4.24 and later')
```
Comment (seando-adsk) [inline test/lib/ufe/testAttributes.py:186]: Should really say something about why the test needs 0.4.24. Such as "Test requires xxx feature only available on Ufe 0.4.24 and later"

**Review by seando-adsk — CHANGES_REQUESTED** [review]

File: test/lib/ufe/testAttributes.py (hunk 2 above)
Comment (alicedegirolamo) [reply test/lib/ufe/testAttributes.py]: I update it.

**Review by JGamache-autodesk — APPROVED** [review]

**Review by seando-adsk — APPROVED** [review]

**Pull request merged**